streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -1606,19 +1664,15 @@ object ZSink extends ZSinkPlatformSpecific {
   /**
    * Creates a single-value sink from a value.
    */
-  final def succeed[B](b: B): ZSink[Any, Nothing, Nothing, Any, B] =
-    new SinkPure[Nothing, Nothing, Any, B] {
-      type State = Unit
-      val initialPure                    = ()
-      def stepPure(state: State, a: Any) = ()
-      def extractPure(state: State)      = Right((b, Chunk.empty))
-      def cont(state: State)             = false
+  final def succeed[A, B](b: B): ZSink[Any, Nothing, A, A, B] =
```

Comment:
I'm guessing this cannot be inferred from the sink before this one? :/

Reply:
No, unfortunately not. Same problem as with `ZStream#run`